service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsController.java

```diff
@@ -40,14 +40,15 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody Call
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
         CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
-        List<Element<Direction>> allPartyDirections = directionHelperService
-            .getDirectionsForAssignee(caseData.getStandardDirectionOrder().getDirections(), ALL_PARTIES);
-
         Map<String, List<Element<Direction>>> sortedDirections =
             directionHelperService.sortDirectionsByAssignee(caseData.getStandardDirectionOrder().getDirections());
 
-        sortedDirections.forEach((assignee, directions) -> directionHelperService
-            .addAssigneeDirectionKeyValuePairsToCaseData(assignee, allPartyDirections, caseDetails));
+        sortedDirections.forEach((assignee, directions) -> {
+            if (!assignee.equals(ALL_PARTIES.getValue())) {
+                directions.addAll(sortedDirections.get(ALL_PARTIES.getValue()));
+                directionHelperService.addAssigneeDirectionKeyValuePairsToCaseData(assignee, directions, caseDetails);
+            }
+        });
```

Comment:
If that would be enum, the code would be a lot nicer with filters and regular streams, but we can leave it.